contracts/SpokePool.sol

```diff
@@ -225,49 +222,130 @@ abstract contract SpokePool is Testable, Lockable, MultiCaller {
         });
         bytes32 relayHash = _getRelayHash(relayData);
 
+        _fillRelay(relayHash, relayData, relayerFeePct, maxTokensToSend, repaymentChain);
+    }
+
+    function _fillRelay(
+        bytes32 relayHash,
+        RelayData memory relayData,
```

Comment:
is there a gas difference between this being `memory` and `calldata`?

Reply:
according to [this answer](https://ethereum.stackexchange.com/questions/74442/when-should-i-use-calldata-and-when-should-i-use-memory), `calldata` is cheaper.

Also aside, when testing I noticed that declaring a variable as `memory` does not contribute to the "stack too deep error", while `calldata` does?

Reply:
Note that `calldata` is only avaialble for `external` methods. [source](https://solidity-by-example.org/data-locations/)